### PR TITLE
Replace usage of defaultProps

### PR DESCRIPTION
**Pull request by gkielwasser**
Replace usage of defaultProps. This is deprecated and will be removed in future React versions (see [facebook/react#16210](https://github.com/facebook/react/pull/16210))

Here is the current warning
```...: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead```

### CODE REVIEW

**Comment by gkielwasser** [comment]
> Reverted due to #140 .
> 
> [facebook/react#16210](https://github.com/facebook/react/pull/16210) deprecated defaultProps on Function Components, but react-lines-ellipsis is still using classes. And there are no warnings as far as I see.

There was indead an issue passing the default props in my pull request. Specially on the class components the reflow was called without the default props. 
```
class LinesEllipsis extends React.PureComponent {
  constructor(props) {
    super(props);
    this.props = _extends({}, defaultProps, props); <--- Wrong
    this.state = {
      text: props.text,
      clamped: false
    };
    this.units = [];
    this.maxLine = 0;
    this.canvas = null;
  }
  componentDidMount() {
    this.initCanvas();
    this.reflow(this.props); <--- default props would not be applied here
  }
```

instead they should have been kept as [static props](https://react.dev/reference/react/Component#static-defaultprops) in the class.

```
class LinesEllipsis extends React.PureComponent {
  static defaultProps = {
    component: 'div',
    text: '',
    lineHeight: 1.2,
    maxLine: 1,
    style: {},
    overflowFallback: false,
  };

  constructor(props) {
    super(props);
    this.state = {
      text: props.text,
      clamped: false,
    };
    this.units = [];
    this.maxLine = 0;
    this.canvas = null;
  }

  componentDidMount() {
    this.initCanvas();
    this.reflow(this.props);
  }
}
```

In React 19 there is no more warning about default props usage. While it will still works for classes it won't work for functional component. Also the functional component `LinesEllipsisLoose` is broken in React 19 because default props are not applied anymore.

To summarize only the functional component should be patched.
